dhis-2/dhis-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/RequestParamsValidatorTest.java

```diff
@@ -353,6 +366,81 @@ void shouldParseFilterWhenFilterHasMultipleOperatorWithFinalColon() throws BadRe
         filters);
   }
 
+  @Test
+  void shouldParseDataElementFilters() throws BadRequestException {
+    Map<UID, List<QueryFilter>> filters =
+        parseDataElementFilters(DE_1_UID + ":lt:20:gt:10," + DE_2_UID + ":like:foo");
+
+    assertEquals(
+        Map.of(
+            DE_1_UID,
+            List.of(
+                new QueryFilter(QueryOperator.LT, "20"), new QueryFilter(QueryOperator.GT, "10")),
+            DE_2_UID,
+            List.of(new QueryFilter(QueryOperator.LIKE, "foo"))),
+        filters);
+  }
+
+  @Test
+  void shouldParseDataElementFilterWhenSingleUnaryOperator() throws BadRequestException {
+    Map<UID, List<QueryFilter>> filters = parseDataElementFilters(DE_1_UID + ":!null");
+
+    assertEquals(Map.of(DE_1_UID, List.of(new QueryFilter(QueryOperator.NNULL))), filters);
+  }
+
+  @Test
+  void shouldParseDataElementFilterWhenMultipleUnaryOperatorsCombined() throws BadRequestException {
+    Map<UID, List<QueryFilter>> filters = parseDataElementFilters(DE_1_UID + ":!null:null");
```

Comment:
that should not be valid

Reply:
see [here](https://dhis2.slack.com/archives/C05S1NTHBJ9/p1739176522707899?thread_ts=1738232925.732909&cid=C05S1NTHBJ9)